### PR TITLE
Make TransformAMDToCJSModule public

### PR DESCRIPTION
**Pull request by MNeise**
Make TransformAMDToCJSModule and its constructor public for adding AMD module support to ClojureScript.

https://groups.google.com/forum/#!topic/closure-compiler-discuss/5VK7IRR8RnI

### CODE REVIEW

File: src/com/google/javascript/jscomp/TransformAMDToCJSModule.java [hunk 1]
```diff
@@ -52,7 +52,13 @@
...
   private int renameIndex = 0;
 
-  TransformAMDToCJSModule(AbstractCompiler compiler) {
+  /**
+   * Creates a new TransformAMDToCJSModule instance which can be used
+   * to convert an AMD module into a CommonJS module.
+   *
+   * @param compiler The compiler
+   */
+  public TransformAMDToCJSModule(AbstractCompiler compiler) {
```
Comment (dimvar) [inline src/com/google/javascript/jscomp/TransformAMDToCJSModule.java]: the jsdoc doesn't add any new information, so you can delete it

**Comment by dimvar** [comment]
lgtm after you delete the jsdoc

File: src/com/google/javascript/jscomp/TransformAMDToCJSModule.java (hunk 1 above)
Comment (MNeise) [inline src/com/google/javascript/jscomp/TransformAMDToCJSModule.java]: yeah, makes sense.

**Comment by MNeise** [comment]
Sure, no problem, deleted the documentation.

**Comment by dimvar** [comment]
Sorry for being a bit late on this. Do you mind also squashing the commits into one commit?

**Comment by MNeise** [comment]
Sure, no problem.

**Comment by MNeise** [comment]
Thank you!

**Pull request merged**